### PR TITLE
Ripple button

### PR DESCRIPTION
**Pull request by ahishek**
- Adding ripple effect on Android buttons
- Change button opacity when state is enabled once it was disabled

### CODE REVIEW

File: src/components/Button/native/Button.js [hunk 1]
```diff
@@ -44,9 +47,38 @@ const Button = styled.TouchableOpacity`
     `,
   }[p.size])}
   ${(p) => p.block ? 'align-self: stretch;' : 'align-self: flex-start;'}
-  ${(p) => p.disabled ? 'opacity: 0.5;' : ''}
+  ${(p) => p.disabled ? 'opacity: 0.5;' : 'opacity: 1'}
   `;
 
+const Button = ({ color, kind, shape, size, block, disabled, children, ...restProps }) => (
+  isPlatformAndroid ?
```
Comment (lakshyaranganath) [inline src/components/Button/native/Button.js]: directly inline the check here, `Platform.OS === 'android' ?`

File: src/components/Button/native/Button.js [hunk 2]
```diff
@@ -44,9 +47,38 @@ const Button = styled.TouchableOpacity`
     `,
   }[p.size])}
   ${(p) => p.block ? 'align-self: stretch;' : 'align-self: flex-start;'}
-  ${(p) => p.disabled ? 'opacity: 0.5;' : ''}
+  ${(p) => p.disabled ? 'opacity: 0.5;' : 'opacity: 1'}
   `;
 
+const Button = ({ color, kind, shape, size, block, disabled, children, ...restProps }) => (
```
Comment (lakshyaranganath) [inline src/components/Button/native/Button.js]: break this into multiple lines

File: src/components/Button/native/Button.js [hunk 3]
```diff
@@ -44,9 +47,38 @@ const Button = styled.TouchableOpacity`
...
+        kind={kind}
+        size={size}
+        shape={shape}
+        block={block}
+        disabled={disabled}
+      >
+        {children}
+      </ButtonView>
+    </TouchableNativeFeedback>
+    :
```
Comment (lakshyaranganath) [inline src/components/Button/native/Button.js]: this is how ternary statements are to be formatted in JSX
```
Platform.OS === 'android' ? (
  <SomeAndroidComponent />
) : (
  <SomeIosComponent />
)
```

File: src/components/Button/native/Button.js [hunk 4]
```diff
@@ -44,9 +47,38 @@ const Button = styled.TouchableOpacity`
     `,
   }[p.size])}
   ${(p) => p.block ? 'align-self: stretch;' : 'align-self: flex-start;'}
-  ${(p) => p.disabled ? 'opacity: 0.5;' : ''}
+  ${(p) => p.disabled ? 'opacity: 0.5;' : 'opacity: 1'}
```
Comment (lakshyaranganath) [inline src/components/Button/native/Button.js]: you can ignore the false statement styling with an empty string
`${(p) => p.disabled ? 'opacity: 0.5;' : ''}`

Comment (ahishek) [reply src/components/Button/native/Button.js]: This doesn't work. For example if a button is disabled and I want to change it to the normal color as soon as its enabled, just passing the empty string doesn't work. I have to reset it back to opacity 1, hence added this.

Comment (lakshyaranganath) [reply src/components/Button/native/Button.js]: I see, must be handled differently on web and native then.

**Comment by lakshyaranganath** [comment]
You seem to have conflicts that need to be resolved.

**Pull request merged**